core/service.go

```diff
@@ -151,18 +162,59 @@ func (s *Service) StorageRoot() (common.Hash, error) {
 	return s.storageState.StorageRoot()
 }
 
+func (s *Service) handleBabeSession() {
+	for {
+		<-s.epochDone
+		log.Trace("CORE: BABE epoch complete, initializing new session")
+
+		newBlocks := make(chan types.Block)
+		s.blkRec = newBlocks
+
+		epochDone := make(chan struct{})
+		s.epochDone = epochDone
+
+		// BABE session configuration
+		bsConfig := &babe.SessionConfig{
+			Keypair:        s.keys[0].(*sr25519.Keypair),
+			Runtime:        s.rt,
+			NewBlocks:      newBlocks, // becomes block send channel in BABE session
+			BlockState:     s.blockState,
+			AuthorityIndex: 0, // TODO: where do we get the BABE authority data?
+			AuthData:       []*babe.AuthorityData{babe.NewAuthorityData(s.keys[0].Public().(*sr25519.PublicKey), 1)},
+			EpochThreshold: big.NewInt(0),
+			Done:           epochDone,
+		}
+
+		// create a new BABE session
+		bs, err := babe.NewSession(bsConfig)
+		if err != nil {
+			log.Error("CORE could not initialize BABE", "error", err)
```

Comment:
Again, not sure `CORE` should be capitalized as it is not an acronym like BABE.

Reply:
can change if you would like, maybe we could discuss later what we should do for consistency?

Reply:
I am ok with it as is if this is the format we want to adopt but maybe we should double check and agree on a format. `CORE` uppercase seems odd to me but might just be me.

Reply:
I've changed it to lowercase for now, I'm personally fine either way as long as it's consistent

Reply:
me too